data-prepper-plugins/map-values-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/mapvalues/MapValuesConfig.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.mapvalues;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class MapValuesConfig {
+
+    @JsonProperty("source")
+    private String source;
+
+    @JsonProperty("target")
+    private String target;
+
+    @JsonProperty("map")
+    private Map<String, String> map;
+    @JsonProperty("file_path")
+    private String filePath;
+
+    @JsonProperty("map_when")
+    private String mapWhen;
+    @JsonProperty("patterns")
+    private HashMap<String, String> patterns;
+
+    @JsonProperty("exact")
+    private Boolean exact;
+    @JsonProperty("regex")
+    @JsonIgnore
+    private Boolean regex;
+
+
+    public String getSource() {
+        return source;
+    }
+
+    public String getTarget() {
+        return target;
+    }
+
+    public Map<String, String> getMap() {
+        return map;
+    }
+
+    public String getFilePath() { return filePath; }
+
+    public String getMapWhen() {
+        return mapWhen;
+    }
+
+    public HashMap<String, String> getPatterns() {
+        return patterns;
+    }
+
+    public Boolean getExact() {
+        return exact;
+    }
+
```

Comment:
You should have validations like either `map` or `regex` is present. We should not accept config that has both of these missing.  `patterns` and `regex` should be made config options under `regex`. You can see other config files on examples to how to make these options under `regex`

Reply:
**You should have validations like either `map` or `regex` is present** --> Yes. Also I need to make `source` & `target` options to be mandatory aslo. 

**`patterns` and `regex` should be made config options under regex** --> I will add the `patterns` and `exact` as config options under `regex`.